test/tools/router_check/router.cc

```diff
@@ -76,43 +71,31 @@ bool RouterCheckTool::compareEntriesInJson(const std::string& expected_route_jso
     }
     Json::ObjectSharedPtr validate = check_config->getObject("validate");
 
-    const std::unordered_map<std::string, std::function<bool(ToolConfig&, const std::string&)>>
-        checkers = {
-            {"cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareCluster(tool_config, expected);
-             }},
-            {"virtual_cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualCluster(tool_config, expected);
-             }},
-            {"virtual_host_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualHost(tool_config, expected);
-             }},
-            {"path_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewritePath(tool_config, expected);
-             }},
-            {"host_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewriteHost(tool_config, expected);
-             }},
-            {"path_redirect",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRedirectPath(tool_config, expected);
-             }},
-        };
+    using checkerFunc = std::function<bool(ToolConfig&, const std::string&)>;
+    const std::unordered_map<std::string, checkerFunc> checkers = {
+        {"cluster_name",
+         [this](auto&... params) -> bool { return this->compareCluster(params...); }},
```

Comment:
Is there any reason to change the code that was already here for this? Generally prefer less magical C++ and not sure the parameter pack passing is worth it here?

Reply:
Not really, I just thought it looked nicer than the full blown function signature on every line.

By worth it do you mean in a performance sense or readability sense? Regarding perf, it should all be sorted out at compile time. Regarding readability, I just thought it looked cleaner.

If you feel strongly against it, I can just revert this block to the original.

Reply:
readability. In general I think the more verbose version is more readable since it's clearer what is being passed, but I don't feel strongly about it.